xopt/generators/bayesian/bax_generator.py

```diff
@@ -0,0 +1,123 @@
+# +
+import logging
+import time
+from typing import Callable, Dict, List
+
+from botorch.optim import optimize_acqf
+from botorch.optim.initializers import sample_truncated_normal_perturbations
+
+from xopt.generators.bayesian.bax.acquisition import ExpectedInformationGain
+
+from xopt.generators.bayesian.bayesian_generator import BayesianGenerator
+from xopt.generators.bayesian.objectives import create_mc_objective
+from xopt.generators.bayesian.options import BayesianOptions
+from xopt.vocs import VOCS
+
+logger = logging.getLogger()
+
+
+class BAXGenerator(BayesianGenerator):
+    alias = "BAX"
+
+    def __init__(
+        self,
+        vocs: VOCS,
+        meas_param: str,
+        algo_class: Callable,
+        algo_kwargs: Dict,
+        options: BayesianOptions = None,
+    ):
+        """
+        Generator using Bayesian Algorithm Execution
+        via Expected Information Gain acquisition function.
+
+        Parameters
+        ----------
+        vocs: dict
+            Standard vocs dictionary for xopt
+
+        options: BayesianOptions
+            Specific options for this generator
+        """
+        options = options or BayesianOptions()
+        if not isinstance(options, BayesianOptions):
+            raise ValueError("options must be a `BayesianOptions` object")
+
+        if vocs.n_objectives != 1:
+            raise ValueError("vocs must have one objective for optimization")
+
+        super().__init__(vocs, options)
+        self.algo = algo_class(**algo_kwargs)
+        self.meas_param = meas_param
+
+    @staticmethod
+    def default_options() -> BayesianOptions:
+        return BayesianOptions()
+
+    def generate(self, n_candidates: int) -> List[Dict]:
```

Comment:
take a look at the base bayesian generator class, I've made some changes to modularize the generation function so I don't think you need to overwrite it, just overwrite the individual components